javascript/app.js

```diff
@@ -70,53 +82,171 @@ FinancialFreedom.controller('bodyController', function($scope, $rootScope, $loca
             amount = 0;
+            return;
+        }
+        
+        $scope.preconvert = false;
+
+        if ($scope.calc_values.useCustomVals) {
+
+            customVals = { // Median American values
+                monthly_income: 4328.25,
+                total_assets: 0,
+                monthly_expenses: 4111.84,
+                income_increase_rate: 0.03,
+                growth_rate: 0.08,
+                expenses_increase_rate: 0.03
+            };  
+        }
+
+        else {
+            customVals = null;
         }
 
-        dates = DollarsToTimeService.calculateDollarsToTime(amount, $scope.calc_values.expense, $scope.calc_values.recurring);
+        var newParams = {
+            amount: amount,
+            expense: $scope.calc_values.expense,
+            recurring: $scope.calc_values.recurring,
+            useCustomVals: $scope.calc_values.useCustomVals
+        };
+
+        var stateOptions = { 
+            location: true, 
+            inherit: true, 
+            relative: $state.$current, 
+            notify: false 
+        };
+
+        $state.transitionTo('dollars-to-time', newParams, stateOptions);  
+
+        outputDate(amount, customVals, newParams);
+    };
+
+    if (params.expense !== undefined) {
+        var processedParams = processParams(params);
+        $scope.calc_values = processedParams;
+    }
+
+    else {
+        $scope.calc_values = default_calc_values;
+    }
+
+    function outputDate(amount, customVals, newParams) {
+        
+        dates = DollarsToTimeService.calculateDollarsToTime(amount, $scope.calc_values.expense, $scope.calc_values.recurring, customVals);
 
         if (dates.difference) {
             $scope.dates.years = dates.difference.years;
             $scope.dates.months = dates.difference.months;
             $scope.dates.days = dates.difference.days;
         }
+    }
+
+    $scope.showYou = function() {
+        if (!UserDataCache.userData.monthly_expenses) {
+            $location.path('/income');
+            DollarsToTimeService.redirectToConverter = true;
+        }
+        else {
+            $scope.switchInputVals('you');
+        }
+    };
+
+    $scope.switchInputVals = function(sourceElement) {
+        if ((sourceElement == 'you') && ($scope.calc_values.useCustomVals === true)) {
+            $scope.calc_values.useCustomVals = false;
+        }
+
+        if ((sourceElement == 'american') && ($scope.calc_values.useCustomVals === false)) {
+            $scope.calc_values.useCustomVals = true;
+        }
+    };
+
+    $scope.$watch('calc_values', function(new_value, old_value) {
+
+        $scope.convert();
+
+        $scope.cashflowLabel = $scope.calc_values.expense ? 'Added to' : 'Reduced from';
+
+        if (new_value !== old_value) {
+
+            $scope.dates = {
+                years: '-',
+                months: '-',
+                days: '-'
+            };
+
+            $scope.preconvert = true;
+
+        }
+

```

Comment:
There's a lot of extra blank lines in this block of code. It's not bad, perse, but some of it can be condensed.